src/components/App.jsx

```diff
@@ -1,74 +1,15 @@
-import React ,{Component}  from "react";
-
-import ReactDOM from "react-dom";
-import Search from "./Search";
-import Login from "./Login";
-import Signup from "./Signup"
-import MovieRow from "./MovieRow";
-import $ from "jquery";
-import {
-  BrowserRouter as Router,
-  Switch,
-  Route,
-  Link
-} from "react-router-dom";
-
-
-class App extends Component {
-
-   constructor(props){
-     super(props)
-      console.log("This is my initializer");
-      this.state={rows:""}
-    //   const movies = [{id:"0",posterPath:"https://m.media-amazon.com/images/M/MV5BZTYzMjA2M2EtYmY1OC00ZWMxLThlY2YtZGI3MTQzOWM4YjE3XkEyXkFqcGdeQXVyMTQxNzMzNDI@._V1_.jpg",title:"Avengers: Infinity war",overview:"zcbjdbfjdbjvnjdbvhbdhbdbnbfnbvjmnbvncmnvmc"},
-    //                   {id:"1",posterPath:"https://m.media-amazon.com/images/M/MV5BZTYzMjA2M2EtYmY1OC00ZWMxLThlY2YtZGI3MTQzOWM4YjE3XkEyXkFqcGdeQXVyMTQxNzMzNDI@._V1_.jpg",title:"The Avengers",overview:"ppppppppppppppppppppp"}];
-    //
-    //   const movieRows =[];
-    //   movies.forEach((movie)=>{
-    //   console.log(movie.overview);
-    //
-    //   const movieRow= <MovieRow movie = {movie}/>
-    //   movieRows.push(movieRow)
-    //
-    // });
-    // this.state={rows:movieRows};
-
-  }
-
-
-  performSearch(movie){
```

Comment:
this logic used for the search http request was moved to the search component, as the search component is the responsible for the search feature, it makes sense to have all the code related inside that component, and then we keep the App component clean. When we use a framework/library that uses components, we try to put all the logic related to a component within it, but there is a good read on this topic that I highly recommend you, which is [Container versus Presentational Components](https://medium.com/@dan_abramov/smart-and-dumb-components-7ca2f9a7c7d0) this article was written by Dan Abramov which works at Facebook (or was working I'm not sure) but he created Redux which you will use with react and maybe other tools later so his content is marvelous and I recommend reading everything you can.